pkg/vendir/config/lock_config.go

```diff
@@ -53,6 +53,13 @@ func NewLockConfigFromBytes(bs []byte) (LockConfig, error) {
 	return config, nil
 }
 
+func LockFileExists(path string) bool {
+	if _, err := os.Stat(path); err != nil {
+		return false
+	}
+	return true
+}
+
```

Comment:
There is nothing specific to a config in this function. I'd move it to a more generic utils file (if such exists).

Otherwise, it is worth checking for the exact error, because there [could be several of them](https://pkg.go.dev/os#pkg-variables).

```suggestion
func FileExists(path string) (bool, error) {
	_, err := os.Stat(path)
	if err == nil {
		return true, nil
	}
	if errors.Is(err, fs.ErrNotExist) {
		return false, nil
	}
	return false, err
}

```

Reply:
Fair point - there is not util at this point. Made it part of the SyncOptions struct to keep usage simple, but give it a context specific home.

Reply:
Don't forget to check for the exact error :-)

Reply:
You got it